### DISCUSSION TITLE
Support for batching

### DISCUSSION THREAD

**Discussion by nicolas-gouteux-sonarsource** [micronaut-projects/micronaut-data#2291] [General]
Hi

It is confusing that you guys use batching to denote the ability to amend several columns in one update/insert.
In classical JDBC parlance, batching refers to batching rows for updates, the user deciding when its time to flush the batch.
I was about to switch from JDBC to Micronaut, but without this feature, it is unusable for me.

Thanks

**Comment by dstepanov** [comment]
Sorry, I don't understand what you mean.
Micronaut Data JDBC makes an ordinary batch statement.
`flush the batch` What do you mean by that? JDBC doesn't have any kind of flushing.

**Reply by nicolas-gouteux-sonarsource** [reply]
Hi 

Either I am too old, or you are too young! 
But there is definitely batching in JDBC since a long time!
Take a look at this
https://jenkov.com/tutorials/jdbc/batchupdate.html#:~:text=A%20JDBC%20batch%20update%20is,for%20each%20one%20to%20finish.

Thanks!

**Reply by dstepanov** [reply]
Micronaut Data JDBC does the batching. Can you post an example that doesn't?

**Reply by nicolas-gouteux-sonarsource** [reply]
Say I want to update 100 rows in one single network back and forth.
In JDBC, you have the sequence:

```
for (i=1 to 100) {
  ps.setInt(1,x); ps.setString(2,y); etc...
  ps.addBatch():
}
ps.executeBatch()
```

I am not sure on how to that in Micronaut...

**Reply by dstepanov** [reply]
In Micronaut Data JDBC you can simply use `myRepository.saveAll(entities)`;

**Reply by raghu298** [reply]
@dstepanov what about just like spring batch functionalities like where you can define a pipeline of readers, writers, processors, chunking, batching, retry like that is anything available in micronaut?

**Reply by dstepanov** [reply]
This question is about JDBC batching
